funnel/views/project.py

```diff
@@ -157,18 +157,22 @@ def json(self):
             .order_by(db.desc('created_at'))
             .all()
         )
-        return jsonp(
-            **{
-                'project': project_data(self.obj),
-                'space': project_data(
-                    self.obj
-                ),  # TODO: Remove when the native app switches over
-                'venues': [venue_data(venue) for venue in self.obj.venues],
-                'rooms': [room_data(room) for room in self.obj.rooms],
-                'proposals': [proposal_data(proposal) for proposal in proposals],
-                'schedule': schedule_data(self.obj),
-            }
-        )
+        # return JSON if project is published
+        if self.obj.state.value == 1:
```

Comment:
1. Don't use raw values. They are hard to track down when upgrading the list of states.
2. This sort of gate should be in a view decorator, not inside the view. In this case, there is a `@requires_permission('view')`. That permission should not be granted if the user is not supposed to see this. Code for that is in the `permissions` method on the model.